File: CybtansSDK/Graphics/Cybtans.Graphics/Lights/AmbientLight.cs
using Cybtans.Math;
using System;
using System.Collections.Generic;
using System.Text;

namespace Cybtans.Graphics.Lights
{
    public class AmbientLight
    {
        public AmbientLight()
        {
            AmbientColor = new Color3(0.1f, 0.1f, 0.1f);
            SkyColor = new Color3(1, 1, 1);
            GroundColor = new Color3(0, 0, 0);
            NorthPole = Vector3.UnitY;
            Intensity = 1;
        }

        public Color3 AmbientColor { get; set; }

        public Color3 SkyColor { get; set; }

        public Color3 GroundColor { get; set; }

        public Vector3 NorthPole { get; set; }

        public float Intensity { get; set; }
    }
}

File: CybtansSDK/Tests/react-app/src/cybtans.graphics/AmbientLight.ts
import { vec3 } from "gl-matrix";
import { float3 } from "./MathUtils";
import { AmbientLightDto } from "./models";

export class AmbientLight {
    ambientColor?: vec3;
    skyColor?: vec3;
    groundColor?: vec3;
    intensity: number;
    northPole?: vec3;

    constructor(data?:AmbientLightDto|null){
        this.ambientColor = float3(data?.ambientColor || [0.1,0.1,0.1]);
        this.skyColor = float3(data?.skyColor || [0.3, 0.3, 0.3]);
        this.groundColor = float3(data?.groundColor || [0.1,0.1,0.1]);
        this.intensity = data?.intensity || 1;
        this.northPole = float3(data?.northPole || [0,1,0]);
    }
}